src/SortByGender.js

```diff
@@ -0,0 +1,30 @@
+import React, { useState } from 'react'
+
+const SortByGender = ({ sortedNames }) => {
+    const [sortByBoys, setSortByBoys] = useState([])
+    const [sortByGirls, setSortByGirls] = useState([])
+    const [sortByAll, setSortByAll] = useState([])
+
+    const handelBoys = () => {
+        setSortByBoys(sortedNames.filter((boy) => boy.sex === "m" ? boy.name : null))
+    }
+    const handelGirls = () => {
+        setSortByGirls(() => sortedNames.filter((girl) => girl.sex === "f" ? girl.name : null))
+    }
+    const handelAll = () => {
+        setSortByAll(sortedNames.filter((all) => all.name));
```

Comment:
When you're handling clicks, you need to update the state you used in the other component to display your favourite list.

Reply:
Also, you probably won't need:
` const [sortByBoys, setSortByBoys] = useState([]);
  const [sortByGirls, setSortByGirls] = useState([]);
  const [sortByAll, setSortByAll] = useState([]);`

Reply:
Could you correct your typos as well? like "Rest Names"

Reply:
Thanks Shadi... now time to solve this, 

I'll come back to you if I stack again thank you